src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceDataRegistryKey.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Buffers;
+using System.IO;
+using Internal.Win32.SafeHandles;
+
+namespace System.Diagnostics
+{
+    internal class PerformanceDataRegistryKey
+    {
+        private const int MaxArrayPoolLength = 1024 * 1024;
```

Comment:
Why is this needed?  It seems like if you exceed this you just fall back to allocating, which is exactly what the shared array pool does when its max size is exceeded.

Reply:
Many thanks!